LearningMissionSimulation/PlaygroundGavril.cs

```diff
@@ -243,6 +255,30 @@ public void RegisterStudentsForClasses()
 
             ReportSummary(actionName: action, itemCount: classroomItemListCount);
             ReportFooter(actionName: action);
+        }
+
+        private void UpdateStudentList(Classroom classroom)
+        {
+            int itemListCount = AttributeGenerator.random.Next(0, classroom.MaximumCapacity - classroom.ItemList.Count + 1);
+            int i = 0;
+            while (i < itemListCount)
```

Comment:
You loop twice here. This is wrong. Your inner loop (**foreach (var student in activeStudentList))** does not check how many students already in the class and allows to register ALL active students.

Reply:
Done

Reply:
DONE